casr/src/bin/casr-java.rs

```diff
@@ -0,0 +1,153 @@
+use casr::util;
+use libcasr::constants::*;
+use libcasr::exception::Exception;
+use libcasr::init_ignored_frames;
+use libcasr::java::*;
+use libcasr::report::CrashReport;
+use libcasr::stacktrace::*;
+
+use anyhow::{bail, Context, Result};
+use clap::{Arg, ArgAction, ArgGroup};
+use regex::Regex;
+use std::path::PathBuf;
+use std::process::Command;
+
+fn main() -> Result<()> {
+    let matches = clap::Command::new("casr-java")
+        .version(clap::crate_version!())
+        .about("Create CASR reports (.casrep) from java reports")
+        .term_width(90)
+        .arg(
+            Arg::new("output")
+                .short('o')
+                .long("output")
+                .action(ArgAction::Set)
+                .value_parser(clap::value_parser!(PathBuf))
+                .value_name("REPORT")
+                .help(
+                    "Path to save report. Path can be a directory, then report name is generated",
+                ),
+        )
+        .arg(
+            Arg::new("stdout")
+                .action(ArgAction::SetTrue)
+                .long("stdout")
+                .help("Print CASR report to stdout"),
+        )
+        .group(
+            ArgGroup::new("out")
+                .args(["stdout", "output"])
+                .required(true),
+        )
+        .arg(
+            Arg::new("stdin")
+                .long("stdin")
+                .action(ArgAction::Set)
+                .value_parser(clap::value_parser!(PathBuf))
+                .value_name("FILE")
+                .help("Stdin file for program"),
+        )
+        .arg(
+            Arg::new("ignore")
+                .long("ignore")
+                .action(ArgAction::Set)
+                .value_parser(clap::value_parser!(PathBuf))
+                .value_name("FILE")
+                .help("File with regular expressions for functions and file paths that should be ignored"),
+        )
+        .arg(
+            Arg::new("ARGS")
+                .action(ArgAction::Set)
+                .num_args(1..)
+                .last(true)
+                .help("Add \"-- <path> <arguments>\" to run"),
+        )
+        .get_matches();
+
+    init_ignored_frames!("java"); //TODO
```

Comment:
What is left to do?

Reply:
Don't we, also, need cpp here for native extensions? I suppose we have a similar bug in `casr-python`.

Reply:
i don't think so, for crashes in cpp code `casr-san` will be called